homework/tests/example.spec.js

```diff
@@ -14,18 +15,165 @@ const data = [
+  test.describe(version + ': Concatenate', () => {
+
+    // Concatenating strings
+    test('Concatenating a and b should result in ab', async ({ page }) => {
+      let calculatorPage = new CalculatorPage(page);
+      await calculatorPage.navigate();
+      await calculatorPage.versionSelect(version);
+      await calculatorPage.fillNumberFields('a', 'b');
+      await calculatorPage.selectOperation('Concatenate');
+      await calculatorPage.calculateResult();
+      await expect(page.locator('#numberAnswerField'), 'Incorrect answer. It should be "ab"').toHaveValue('ab');
+    });
+
+    // Concatenating integers
+    test('Concatenating 3 and 4 should result in 34', async ({ page }) => {
+      let calculatorPage = new CalculatorPage(page);
+      await calculatorPage.navigate();
+      await calculatorPage.versionSelect(version);
+      await calculatorPage.fillNumberFields('3', '4');
+      await calculatorPage.selectOperation('Concatenate');
+      await calculatorPage.calculateResult();
+      await expect(page.locator('#numberAnswerField'), 'Incorrect answer. It should be "34"').toHaveValue('34');
+    });
+  });
+
+  // Add functionality
   test.describe(version + ': Add', () => {
-    test('Concatenating 2 and 3 results in 23', async ({ page }) => {
-      await page.goto('https://testsheepnz.github.io/BasicCalculator');
-      await page.selectOption('#selectBuild', { label: version});
-      await page.locator('#number1Field').type('2');
-      await page.locator('#number2Field').type('3');
-      await page.selectOption('#selectOperationDropdown', {label: 'Add'});
-      await page.locator('#calculateButton').click();
-  
-      await expect(page.locator('#numberAnswerField')).toHaveValue('5');
+    test('Adding 3 and 4 should result in 7', async ({ page }) => {
+      let calculatorPage = new CalculatorPage(page);
+      await calculatorPage.navigate();
+      await calculatorPage.versionSelect(version);
+      await calculatorPage.fillNumberFields('3', '4');
+      await calculatorPage.selectOperation('Add');
+      await calculatorPage.calculateResult();
+      await expect(page.locator('#numberAnswerField'), 'The answer is incorrect. It should be 7').toHaveValue('7');
+    });
+  });
+
+  // Subtract functionality
+  test.describe(version + ': Subtract', () => {
+    test('Subtracting 5 and 2 should result in 3', async ({ page }) => {
+      let calculatorPage = new CalculatorPage(page);
+      await calculatorPage.navigate();
+      await calculatorPage.versionSelect(version);
+      await calculatorPage.fillNumberFields('5', '2');
+      await calculatorPage.selectOperation('Subtract');
+      await calculatorPage.calculateResult();
+      await expect(page.locator('#numberAnswerField'), 'The answer is incorrect. It should be 3').toHaveValue('3');
+    });
+  });
+
+  // Multiply functionality
+  test.describe(version + ': Multiply', () => {
+    test('Multiplying 5 and 2 should result in 10', async ({ page }) => {
```

Comment:
Optional.
Please avoid specific values in test name.
Rather use their types -> `Adding integer and string displays an error`

In this case, if we change what values we enter in the test, we do not need to revisit the test name, so they would match.